### PR TITLE
Streetcode search

### PR DESCRIPTION
**Pull request by MaksBrat**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/features/StreetcodePage/Streetcode.styles.scss [hunk 1]
```diff
@@ -39,3 +41,17 @@ $bgImg: '@assets/images/streetcode-card/background.png';
...
     }   
 }
+
+.loader-container{
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background-color: #F4F4F4;
```
Comment (Dobriyr) [inline src/features/StreetcodePage/Streetcode.styles.scss:51]: and here)

**Review by Dobriyr — APPROVED** [review]

File: src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.component.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import './SearchItem.styles.scss';
+
+import { StreetcodeFilterResultDTO } from '@/models/filters/streetcode-filter.model';
+
+const BASE_URL = 'http://localhost:3000/streetcode/';
```
Comment (Katerix) [inline src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.component.tsx:5]: this is already available in other files, just import it

File: src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.styles.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+.resultItemContainer{ 
+    max-width: f.pxToRem(915px);
+    @include mut.flexed($align-items: center);
+    padding: f.pxToRem(13px) f.pxToRem(20px);
+
+    white-space: nowrap;
+
+    border-bottom: 1px solid c.$accented-gray-color;
+    &:last-child{
```
Comment (Katerix) [inline src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.styles.scss:14]: i would add an empty line between lines 13 &14

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx [hunk 4]
```diff
@@ -62,7 +64,9 @@ const RelatedFigureItem = ({ relatedFigure, setActiveTagId, filterTags = true, h
                             {
                                 alias !== null ? (
                                     <p className="aliasText">
-                                        ({alias})
+                                        (
+                                        {alias}
+)
                                     </p>
```
Comment (Katerix) [inline src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx:70]: why

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx [hunk 5]
```diff
@@ -101,7 +105,9 @@ const RelatedFigureItem = ({ relatedFigure, setActiveTagId, filterTags = true, h
...
```
Comment (Katerix) [inline src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx:111]: why x2

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigures.component.tsx [hunk 6]
```diff
@@ -89,8 +89,8 @@ const RelatedFiguresComponent = ({ setActiveTagId } : Props) => {
         swipeOnClick: false,
         slidesToShow: windowsize.width > 1024 ? 4 : windowsize.width <= 480 ? 2 : undefined,
         slidesToScroll: windowsize.width > 1024 ? undefined : windowsize.width <= 480 ? 1 : 3,
-        rows: 1
-    }; 
+        rows: 1,
+    };
```
Comment (Katerix) [inline src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigures.component.tsx:93]: coma is not necessary here

File: src/features/StreetcodePage/TextBlock/TextBlock.component.tsx [hunk 7]
```diff
@@ -37,7 +40,7 @@ const TextComponent = () => {
                 </div>
             </div>
             <div className="videoComponent">
-                <VideoPlayer videoUrls={String(video?.url.href)} />
+                <VideoPlayer videoUrls={String(video?.url.href)} setTextBlockState={setTextBlockState} />
                 {/* <Video videoUrls={"f55dHPEY-0U"}/> */}
```
Comment (Katerix) [inline src/features/StreetcodePage/TextBlock/TextBlock.component.tsx:44]: what is this comment for?

**Review by Katerix — COMMENTED** [review]
Well done! have some questions tho

File: src/features/StreetcodePage/TextBlock/TextBlock.component.tsx (hunk 7 above)
Comment (MaksBrat) [reply src/features/StreetcodePage/TextBlock/TextBlock.component.tsx:44]: i dont know, it`s not mine

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx (hunk 4 above)
Comment (MaksBrat) [reply src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx:70]: eslint did this

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx (hunk 5 above)
Comment (MaksBrat) [reply src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx:111]: eslint did this

File: src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.styles.scss (hunk 3 above)
Comment (MaksBrat) [reply src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.styles.scss:14]: done

File: src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx (hunk 5 above)
Comment (Katerix) [reply src/features/StreetcodePage/RelatedFiguresBlock/RelatedFigureItem/RelatedFigureItem.component.tsx:111]: bad eslint :(

File: src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.component.tsx (hunk 2 above)
Comment (MaksBrat) [reply src/app/layout/header/SearchBlock/SearchResultItem/SearchItem.component.tsx]: done

**Pull request merged**